src/shared/condition.c

```diff
@@ -265,6 +267,61 @@ static int condition_test_kernel_version(Condition *c, char **env) {
         return true;
 }
 
+static int condition_test_osrelease(Condition *c, char **env) {
+        const char *parameter = c->parameter;
+        int r;
+
+        assert(c);
+        assert(c->parameter);
+        assert(c->type == CONDITION_OS_RELEASE);
+
+        for (;;) {
+                _cleanup_free_ char *key = NULL, *condition = NULL, *actual_value = NULL;
+                OrderOperator order;
+                const char *word;
+                bool matches;
+
+                r = extract_first_word(&parameter, &condition, NULL, EXTRACT_UNQUOTE);
+                if (r < 0)
+                        return log_debug_errno(r, "Failed to parse parameter: %m");
+                if (r == 0)
+                        break;
+
+                /* parse_order() needs the string to start with the comparators */
+                word = condition;
+                r = extract_first_word(&word, &key, "!<=>", EXTRACT_RETAIN_SEPARATORS);
+                if (r < 0)
+                        return log_debug_errno(r, "Failed to parse parameter: %m");
+                /* The os-release spec mandates env-var-like key names */
+                if (r < 1 || isempty(word) || !env_name_is_valid(key))
+                        return log_debug_errno(SYNTHETIC_ERRNO(EINVAL),
+                                        "Failed to parse parameter, key/value format expected: %m");
+
+                /* Do not allow whitespace after the separator, as that's not a valid os-release format */
+                order = parse_order(&word);
+                if (order < 0 || isempty(word) || strspn(word, WHITESPACE) > 0)
+                        return log_debug_errno(SYNTHETIC_ERRNO(EINVAL),
+                                        "Failed to parse parameter, key/value format expected: %m");
+
+                r = parse_os_release(NULL, key, &actual_value);
+                if (r < 0)
+                        return log_debug_errno(r, "Failed to parse os-release: %m");
```

Comment:
so parse_os_release() understands quotes around the value. i.e. I figure people who have `PRETTY_NAME="foo bar"` in their os-release file might expect that `ConditionOSRelease=PRETTY_NAME="foo bar"` actually works too.

I wonder what the right way forward here is. One of the following maybe:

1. Beef up parse_env_file_internal() to make its parser usable for this purpose here, i.e. so that it can parse "fractions" of env files/os-release files too, instead of the whole files. Not sure I am particularly a fan of this idea though.
2. Extract the value with extract_first_word() too, and set EXTRACT_UNQUOTE flag so that we cover quotes. If so, parsing of these condition expressions will be different than parsing the os-release files

i am leaning towards 2. Even though we process env files/os-release files here I don't think we have to follow 1:1 the same parse on it here, since we are in a unit file here after all, and we should rather follow our usual semantics here that shoehorn the env file semantics into unit files here.

Of course when using extract_first_word here the question is what to do if people specify two space separated words on the right hand side without any quotes... probably refuse?

Reply:
But that works already with the current version - as each statement is extracted with EXTRACT_UNQUOTE - there's a test case for it as well (not very obvious since we don't control /etc/os-release, so have to be conservative)

Here's a manual test:

```
root@image:~# systemd-run -t --property 'ConditionOSRelease=PRETTY_NAME="Debian GNU/Linux 11 (bullseye)"' echo hello
Running as unit: run-u12.service
Press ^] three times within 1s to disconnect TTY.
hello
```